### PR TITLE
Add hermite polynomial coefficients

### PR DESCRIPTION
**Pull request by jmh530**
I just have the coefficients here and use `mir.polynomial` to produce the results as needed.

I set them up as `enums` so hopefully it is all CTFE-able. 

I plan on making use of the Hermite polynomials for the Cornish-Fisher expansion in `mir-stat`, but I figured it could have use in more places than just there, which is why I added it to `mir-algorithm`, though I am indifferent as to which project or where in the project it would be.

### CODE REVIEW

**Comment by thewilsonator** [comment]
Are we never going to need more than 10 of them?

**Comment by jmh530** [comment]
I just grabbed what was listed on Wikipedia and figured more could be added
later if anyone needed them. There is a recurrence relation that can be
used to get any polynomial. I can add that for the others if you want.

File: source/mir/math/func/hermite.d [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+/++
+Physicist's Hermite polynomial coefficients
+
+Params:
+    N = Degree of polynomial
+
+See_also:
+    $(LINK2 https://en.wikipedia.org/wiki/Hermite_polynomials, Hermite polynomials)
++/
+template hermiteCoefficients(size_t N)
```
Comment (9il) [inline source/mir/math/func/hermite.d:83]: It is good that it is a general purpose now. Could we make it GC CTFE-able function that returns an array of longs? That will compile much faster then template instantiation but will be still usable in @nogc CTFE code.

Comment (jmh530) [reply source/mir/math/func/hermite.d]: @9il Done.

**Comment by jmh530** [comment]
Thanks, tagged new release.

**Pull request merged**